libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/internal/route/line/MapboxRouteLineUtils.kt

```diff
@@ -382,8 +382,8 @@ object MapboxRouteLineUtils {
     }
 
     private fun getRoadClassForIndex(roadClassArray: Array<String?>, index: Int): String? {
-        return if (roadClassArray.size > index) {
-            roadClassArray.slice(0..index).last { it != null }
+        return if (roadClassArray.isNotEmpty() && roadClassArray.size > index && index >= 0) {
+            return roadClassArray[index] ?: getRoadClassForIndex(roadClassArray, index - 1)
```

Comment:
there's an extra `return` in here now

Reply:
ah, good catch. Could you add it to https://github.com/mapbox/mapbox-navigation-android/pull/4246 @cafesilencio?

Reply:
added